search/src/main/java/artic/edu/search/SearchResultsManager.kt

```diff
@@ -119,22 +111,103 @@ class SearchResultsManager(private val searchService: SearchServiceProvider,
-        } else {
-            articObjectDao.getObjectsByIdList(artwork.map { it.artworkId.toString() }).toObservable()
+    private fun loadArtwork(artwork: List<ApiSearchContent.SearchedArtwork>?): Observable<List<ArticSearchArtworkObject>> {
+        return articDataObjectDao.getDataObject()
+                //Get the base Server url just in case we need it... which we might
+                .map { dataObject -> dataObject.imageServerUrl }
+                .toObservable()
+                .flatMap {
+                    generateArtworkObjectList(it, artwork)
+                }
+
+    }
+
+    private fun generateArtworkObjectList(baseUrl: String,
+                                          artwork: List<ApiSearchContent.SearchedArtwork>?)
+            : Observable<List<ArticSearchArtworkObject>> {
+
+        //Create new observable to handle merging the list and stuff
+        return Observable.create<List<ArticSearchArtworkObject>> { emitter ->
+            val returnList = mutableListOf<ArticSearchArtworkObject>()
+            artwork?.forEach { searchedArtwork ->
+                val artworkId = searchedArtwork.artworkId.toString()
+                val articObject = articObjectDao.getObjectByIdSyncronous(artworkId)
+                val gallery = articGalleryDao.getGalleryForIdSynchronously(searchedArtwork.gallery_id)
+                if (articObject != null) {
+                    returnList.add(
+                            transformArticObjectToArticSearchObject(
+                                    artworkId,
+                                    articObject,
+                                    gallery
+                            )
+                    )
+                } else if (searchedArtwork.isOnView) {
+                    returnList.add(
+                            transformSearchedArtworkToSearchArtworkObject(
+                                    searchedArtwork,
+                                    baseUrl,
+                                    gallery
+                            )
+                    )
+                }
+
+            }
+
+            emitter.onNext(returnList)
+            emitter.onComplete()
         }
     }
 
+    private fun transformSearchedArtworkToSearchArtworkObject(
+            searchedArtwork: ApiSearchContent.SearchedArtwork,
+            baseUrl: String,
+            gallery: ArticGallery?
+    ): ArticSearchArtworkObject {
+
+        val imageBaseUrl = baseUrl + "/" + searchedArtwork.image_id
+
+        val thumbnailUrl = "$imageBaseUrl/full/!200,200/0/default.jpg"
+        val imageUrl = "$imageBaseUrl/full/!800,800/0/default.jpg"
+
+        return ArticSearchArtworkObject(
+                searchedArtwork.artworkId.toString(),
+                null,
+                searchedArtwork.title,
+                thumbnailUrl,
+                imageUrl,
+                searchedArtwork.artist_display,
+                searchedArtwork.latlon,
+                gallery?.floor ?: 0,
```

Comment:
Any reason not to use `INVALID_FLOOR` constant?

Reply:
we have one? I always thought we default to 0

Reply:
Nah, we had trouble with that at one point since it's the constant for lower level. `INVALID_FLOOR` is in the db module somewhere.

Reply:
ahh